sound/soc/intel/boards/sof_sdw_rt712_sdca.c

```diff
@@ -39,26 +39,28 @@ int rt712_spk_rtd_init(struct snd_soc_pcm_runtime *rtd)
 	struct snd_soc_card *card = rtd->card;
 	int ret;
 
+
+	if (!strstr(card->components, "spk:")) {
```

Comment:
if the other speaker is not rt712 then we will not have a widget for the rt712?

Reply:
> if the other speaker is not rt712 then we will not have a widget for the rt712?

The rt712_spk_controls and rt712_spk_widgets have SOC_DAPM_PIN_SWITCH("Speaker"), and `SND_SOC_DAPM_SPK("Speaker", NULL),` only and they are already added by other speaker codecs.